packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1230,6 +1230,47 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     });
   }
 
+  /// Whether this has been disposed.
+  ///
+  /// If asserts are disabled, this property is always null.
+  bool? get debugDisposed {
```

Comment:
Should there be more methods guarded with an assert on this? e.g. presumably, you shouldn't re-attach a RenderObject that has been disposed?

Reply:
Added to attach, scheduleIniitialPaint, layout, markNeedsCompositingBitsUpdate, markNeedsPaint, replaceRootLayer, scheduleInitialSemantics, markNeedsSemanticsUpdate.

Other things are covered by _debugCanPerformMutations. I was surprised more of these weren't.